client/src/pages/Configurations/ConfigActivate/index.tsx

```diff
@@ -1,55 +1,189 @@
 import { useParams } from 'react-router';
 import { Title } from '../../../components/Title';
-import { NotFound } from '../../NotFound';
 import {
   NavigationContainer,
   SectionContainer,
   TitleContainer,
 } from '../layout';
 import { StepsContainer, Steps } from '../Steps';
 import { ConfigurationTitleBar } from '../ConfigurationTitleBar';
-// import { useGetConfiguration } from '../../../api/configurations/configurations';
-// import { Spinner } from '../../../components/Spinner';
+import {
+  getGetConfigurationQueryKey,
+  useActivateConfiguration,
+  useDeactivateConfiguration,
+  useGetConfiguration,
+} from '../../../api/configurations/configurations';
+import { Spinner } from '../../../components/Spinner';
+import { VersionMenu } from '../ConfigBuild/VersionMenu';
+import { Status } from '../ConfigBuild/Status';
+import { useToast } from '../../../hooks/useToast';
+import { GetConfigurationResponse } from '../../../api/schemas';
+import { QueryClient, useQueryClient } from '@tanstack/react-query';
+import { ActivationButtons } from './ActivationButtons';
 
 export function ConfigActivate() {
   const { id } = useParams<{ id: string }>();
 
-  // const {
-  //   data: response,
-  //   isPending,
-  //   isError
-  // } = useGetConfiguration(id ?? '');
+  const {
+    data: configuration,
+    isPending,
+    isError,
+    isSuccess,
+  } = useGetConfiguration(id ?? '');
+  const showToast = useToast();
+
+  const { mutate: activate } = useActivateConfiguration();
+  const { mutate: deactivate } = useDeactivateConfiguration();
+
+  const queryClient = useQueryClient();
+
+  if (isPending) return <Spinner variant="centered" />;
+  if (!id || isError) return 'Error!';
+
+  function handleActivation() {
+    if (!configuration || !configuration.data) {
+      showToast({
+        variant: 'error',
+        heading: 'Something went wrong',
+        body: 'Condition not defined',
+      });
+      return;
+    }
+    activate(
+      {
+        configurationId: configuration.data.id,
+      },
+      {
+        onSuccess: async () => {
+          await refetchConfigurations(configuration.data, queryClient);
```

Comment:
When you invalidate the query it should auto-refresh the data and make the page appear up to date. Did you run into any issues with this?
```suggestion
          await queryClient.invalidateQueries({
            queryKey: getGetConfigurationQueryKey(configuration.data.id),
          });
```